src/test/java/com/google/sps/servlets/PotentialMatchesServletTest.java

```diff
@@ -0,0 +1,233 @@
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.servlets.PotentialMatchesServlet;
+
+@RunWith(JUnit4.class)
+public class PotentialMatchesServletTest {
+  private static final String NO_POTENTIAL_MATCH_RESULT = "NO_POTENTIAL_MATCHES";
+
+  private static final String USER_ENTITY = "User";
+  private static final String USER_FRIENDS_LIST_PROPERTY = "friends-list";
+  private static final String USER_ID_PROPERTY = "id";
+  private static final String USER_NAME_PROPERTY = "name";
+  private static final String USER_EMAIL_PROPERTY = "email";
+  private static final String USER_BIO_PROPERTY = "bio";
+
+  private static final String TEST_USER_1_NAME = "Daenerys";
+  private static final String TEST_USER_1_ID = "5555";
+  private static final String TEST_USER_1_EMAIL = "targaryen@gmail.com";
+  private static final String TEST_USER_1_BIO = "dragons <3";
+
+  private static final String TEST_USER_2_NAME = "Eliza";
+  private static final String TEST_USER_2_ID = "1776";
+  private static final String TEST_USER_2_EMAIL = "elizaHam@gmail.com";
+  private static final String TEST_USER_2_BIO = "hamilFam";
+
+  private static final String TEST_USER_3_NAME = "Rory";
+  private static final String TEST_USER_3_ID = "1234";
+  private static final String TEST_USER_3_EMAIL = "gilmore@gmail.com";
+  private static final String TEST_USER_3_BIO = "Stars Hallow :)";
+
+  private static final String TEST_USER_4_NAME = "Steve";
+  private static final String TEST_USER_4_ID = "1964";
+  private static final String TEST_USER_4_EMAIL = "capAmerica@gmail.com";
+  private static final String TEST_USER_4_BIO = "Avengers, assemble";
+
+  private final LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Mock
+  private HttpServletRequest mockRequest;
+ 
+  @Mock
+  private HttpServletResponse mockResponse;
+
+  private PotentialMatchesServlet servletUnderTest;
+  private DatastoreService datastore;
+ 
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    servletUnderTest = new PotentialMatchesServlet();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+  * Tests scenario where there is only one user in datastore
+  *
+  * <p>Should result in no next potential match found
+  */
+  @Test
+  public void oneUserNoPotentialMatches() throws Exception {
+    when(mockRequest.getParameter("userid")).thenReturn(TEST_USER_1_ID);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(mockResponse.getWriter()).thenReturn(writer);
+
+    addTestUserEntityToDatastore(datastore, TEST_USER_1_ID, TEST_USER_1_NAME,
+      TEST_USER_1_EMAIL, TEST_USER_1_BIO, new String[]{});
+    
+    servletUnderTest.doGetWrapper(datastore, mockRequest, mockResponse);
```

Comment:
I think a general test structure would be fairly clean like this:

```
public void twoUserNameThings() {
  addTestUserEntityToDatastore(...);
  addTestUserEntityToDatastore(...);  // as needed, of course

  String nextMatch = execute(TEST_USER_1_ID);

  assertThat(nextMatch).isEqualTo(NO_POTENTIAL_MATCH_RESULT);
}

private String execute(String userIdToFetch) {
  when(mockRequest.getParameter(USER_ID_URL_PARAM)).thenReturn(userIdToFetch);

  servletUnderTest.doGetWrapper(datastore, mockRequest, mockResponse);
  return responseWriter.toString();
}
```

Reply:
Got it! I had made some modifications to the servlet response so that it returns a JSON object, and I also removed the doGetWrapper, so I used this format you suggested but with some changes to account for these modifications.